cmd/state-svc/internal/rtwatcher/watcher.go

```diff
@@ -98,6 +105,39 @@ func (w *Watcher) check() {
 func (w *Watcher) RecordUsage(e entry) {
 	logging.Debug("Recording usage of %s (%d)", e.Exec, e.PID)
 	w.an.Event(anaConst.CatRuntimeUsage, anaConst.ActRuntimeHeartbeat, e.Dims)
+	w.UpdateLastUsed(e)
+}
+
+func (w *Watcher) UpdateLastUsed(e entry) {
+	logging.Debug("Updating last usage of project that contains %s", e.Exec)
+	localProjects := projectfile.GetProjectMapping(w.cfg)
+	for namespace, checkouts := range localProjects {
+		for _, checkout := range checkouts {
+			proj, err := project.FromPath(checkout)
+			if err != nil {
+				multilog.Error("Unable to get project %s from checkout: %v", checkout, err)
```

Comment:
```suggestion
```

This would almost certainly have to be some form of permission error that we can't help with. `GetProjectMapping()` cleans up the projects before returning them, so you should only get valid ones to begin with.

I'm concerned since this is happening in a loop that runs periodically that it could end up spamming our rollbar. Which shouldn't be an issue but sadly it is due to rollbar quotas.